Assets/Scripts/Mecha/Gunner/CannonBehavior.cs

```diff
@@ -43,9 +51,29 @@ void Update ()
             EventManager.onGunnerShot.Invoke(ShotType.Missile);
             // for the scoring script
             EventManager.onGunnerConsumesEnergy.Invoke(MissileEnergyConsumption);
-            Instantiate(MissileShot, Muzzle.position, Muzzle.rotation);
+            projectile = Instantiate(MissileShot, Muzzle.position, Muzzle.rotation);
             mechaController.ConsumeEnergy(MissileEnergyConsumption);
         }
+
+        Ray ray = cam.ScreenPointToRay(Input.mousePosition);
+       
+        if (Physics.Raycast(ray, out hitInfo, aimRange))
+        {
+            transformToAim = hitInfo.point;
+        }
+        else
+        {
+            transformToAim = Vector3.zero;
+        }
+      
+        if (!(transformToAim == Vector3.zero) && projectile != null)
```

Comment:
* `!(transformToAim == Vector3.zero)` => `transformToAim  != Vector3.zero`
 * you can put this condition in the line 60